pkg/basecontroller/factory/factory.go

```diff
@@ -31,7 +31,7 @@ type Factory struct {
 // Informer represents any structure that allow to register event handlers and informs if caches are synced.
 // Any SharedInformer will comply.
 type Informer interface {
-	AddEventHandler(handler cache.ResourceEventHandler)
+	AddEventHandler(handler cache.ResourceEventHandler) (cache.ResourceEventHandlerRegistration, error)
```

Comment:
I think this will not work with sharedinformer. Maybe we should have another interface that just for informer from controller-runtime

Reply:
@qiujian16 `SharedInformer` has been updated too, see https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L143